vendor/github.com/projectcontour/contour/internal/dag/accessors.go

```diff
@@ -78,6 +83,38 @@ func (dag *DAG) EnsureService(meta types.NamespacedName, port intstr.IntOrString
 	return dagSvc, nil
 }
 
+func validateExternalName(svc *v1.Service, enableExternalNameSvc bool) error {
+
+	// If this isn't an ExternalName Service, we're all good here.
+	en := externalName(svc)
+	if en == "" {
+		return nil
+	}
+
+	// If ExternalNames are disabled, then we don't want to add this to the DAG.
+	if !enableExternalNameSvc {
```

Comment:
Since DomainMapping's HTTProxy points to the ExternalName service, we need to set `enableExternalNameSvc` to true.
The current failure of prow is due to this.

Reply:
Ugh, I didn't test that manually.

If we set `enableExternalNameSvc` to true, we're effectively defeating much of the CVE protection. 😢

Reply:
Just to say this out loud (not suggesting we hack this change in here since we'd be forking contour that way, obviously, but maybe it triggers a better idea in someone - or _maybe_ we can suggest this upstream), a potentially safe thing to do might be to allow targets of the form "name.ns.svc.cluster.local" (ie allowlist rather than denylist). That would - I think - work fine for our use cases and I don't _think_ there's any way those can resolve to localhost since that dns space is managed by k8s.

Reply:
> we're effectively defeating much of the CVE protection

Doesn't the `localhostNames` check below resolve that?

Reply:
Nm - it was pointed out to me you could have a non-obvious ExternalName that resolves to 127.0.0.1